src/data_gradients/managers/classification_manager.py

```diff
@@ -0,0 +1,108 @@
+from typing import Optional, Iterable, Callable, List
+
+import torch
+
+from data_gradients.batch_processors.classification import ClassificationBatchProcessor
+from data_gradients.config.data.data_config import ClassificationDataConfig
+from data_gradients.config.data.typing import SupportedDataType, FeatureExtractorsType
+from data_gradients.config.utils import get_grouped_feature_extractors
+from data_gradients.managers.abstract_manager import AnalysisManagerAbstract
+
+
+class ClassificationAnalysisManager(AnalysisManagerAbstract):
+    """Implementation of analysys manager for image classification task.
+    Definition of task name, task-related preprocessor and parsing related configuration file
+    """
+
+    def __init__(
+        self,
+        *,
+        report_title: str,
+        train_data: Iterable,
+        val_data: Optional[Iterable] = None,
```

Comment:
Actually, do we really support `val_data=None`? With other tasks we don't fully support it, so if it's also the case for classification let's remove `Optional`

Reply:
We do. For classification task this is supported.